apple/swift-docc
# [Feature] Emit a warning for a "See Also" task group within a Topics section (SR-15317)

# Feature
https://bugs.swift.org/browse/SR-15317

## Summary

Add a new checker - SeeAlsoInTopicsHeadingChecker to check this situation

It will give a warning and a fixit hint as the SR describe

<img width="991" alt="截屏2021-10-16 20 44 52" src="https://user-images.githubusercontent.com/43724855/137588168-b5cd8bb8-19bf-43d7-a955-17c977ce8fd9.png">

## Testing

Steps:
1. Create a new Xcode project using Xcode 13, and add the documentation.
2. Use the "wrong" markdown described in the SR-15317
3. Change the DocC version Xcode use as the instruction below

(Somehow, the default flag Xcode use seems ignore the warning severity, you need to manually change this Problem's severity from warning to error to test it on Xcode)

> ### Installing into Xcode
>
> You can test a locally built version of Swift-DocC in Xcode 13 or later by setting the DOCC_EXEC build setting to the path of your local docc:
> 
> Select the project in the Project Navigator.
>
> In the Build Settings tab, click '+' and then 'Add User-Defined Setting'.
>
> Create a build setting DOCC_EXEC with the value set to /path/to/docc.
>
> The next time you invoke a documentation build with the "Build Documentation" button in Xcode's Product menu, your custom docc will be used for the build. You can confirm that your custom docc is being used by opening the latest build log in Xcode's report navigator and expanding the "Compile documentation" step.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
